### PR TITLE
Accelerator API docs

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Adds docs for the Accelerator API

cc @edenlightning 

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/accelerators/accelerator.py [hunk 1]
```diff
@@ -512,7 +522,9 @@ def setup_optimizers_in_pre_dispatch(self) -> bool:
         Override to delay setting optimizers and schedulers till after dispatch.
         This is useful when the `TrainingTypePlugin` requires operating on the wrapped accelerator model.
         However this may break certain precision plugins such as APEX which require optimizers to be set.
-        Returns: If True, delay setup optimizers till pre_dispatch, else call within setup.
+
+        Returns:
+            If True, delay setup optimizers till pre_dispatch, else call within setup.
```
Comment (ananthsub) [inline pytorch_lightning/accelerators/accelerator.py:527]:
```suggestion
            If True, delay setup optimizers until `pre_dispatch`, else call within `setup`.
```

**Review by ananthsub — APPROVED** [review]

**Comment by kaushikb11** [comment]
I really like this Lightning Structure Slide. We should add this in the docs as well.

**Comment by williamFalcon** [comment]
@awaelchli can we add plugins? how do they work with accelerators?

**Comment by ananthsub** [comment]
Screen Shot 2021-04-09 at 11 30 49 AM
I've got something like this as well. Would be great to have a polished version on the docs site

**Review by edenlightning — APPROVED** [review]

**Comment by williamFalcon** [comment]
let's merge this and add the images in a follow-up PR

**Comment by awaelchli** [comment]
will add plugins api reference in follow up PR, some build errors need to be resolved

File: docs/source/extensions/accelerators.rst [hunk 2]
```diff
@@ -1,10 +1,52 @@
...
+Currently there are accelerators for:
+
+- CPU
+- GPU
+- TPU
+
+Each Accelerator gets two plugins upon initialization:
+One to handle differences from the training routine and one to handle different precisions.
+
+.. code-block:: python
```
Comment (Borda) [inline docs/source/extensions/accelerators.rst:21]:
```suggestion
.. testcode::
```

**Review by Borda — APPROVED** [review]

**Review by kaushikb11 — APPROVED** [review]

**Pull request merged**